src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,83 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { createComment } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  loadComments: (postId: number) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, loadComments }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [textError, setTextError] = useState(false);
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  const changeName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setNameError(false);
+  };
+
+  const changeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setEmailError(false);
+  };
+
+  const changeText = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setText(e.target.value);
+    setTextError(false);
+  };
+
+  const reset = () => {
+    setName('');
+    setEmail('');
+    setText('');
+    setNameError(false);
+    setEmailError(false);
+    setTextError(false);
+  };
+
+  const onSubmitHandler = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setNameError(!name.trim().length);
+    setEmailError(!email.trim().length);
+    setTextError(!text.trim().length);
+
+    if (!name.trim().length || !email.trim().length || !text.trim().length) {
+      return;
+    }
+
+    const newComment: Omit<Comment, 'id'> = {
+      name,
+      email,
+      body: text,
+      postId,
+    };
+
+    setIsSubmiting(true);
+
+    createComment(newComment)
+      .then(() => {
+        loadComments(postId);
+      })
+      .finally(() => {
+        setIsSubmiting(false);
+      });
+
+    reset();
+  };
```

Comment:
No need to load comments, you receive the new comment as an answer from the server, and if the new comment was added successfully, you can just update your comments state. Also, you shouldn't reset all form fields, only comment body here. The full reset is used only after click on reset button